e2e/test/provisioning/ProvisioningServiceClientE2ETests.cs

```diff
@@ -346,15 +365,20 @@ public async Task ProvisioningServiceClient_GroupEnrollments_Create_Ok(string pr
                             AllocationPolicy = allocationPolicy,
                             ReprovisionPolicy = reprovisionPolicy,
                             CustomAllocationDefinition = customAllocationDefinition,
-                            IotHubs = iotHubsToProvisionTo
                         };
+                        foreach (string hub in iotHubsToProvisionTo)
+                        {
+                            individualEnrollment.IotHubs.Add(hub);
```

Comment:
I know I agreed on following the sdk team's recommendation of keeping the collection "readonly", but this does feel like a noticeable loss in convenience to me. What do you think?

Reply:
I think the way we choose to use the API in this case where we have another list of hubs we bring in, does make it difficult. I don't expect most people would use it that way, though.

Reply:
I'm clarifying with Krzysztof

Reply:
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections#collection-properties-and-return-values

> DO NOT return null values from collection properties or from methods returning collections. Return an empty collection or an empty array instead.

> DO NOT provide settable collection properties.
>
> Users can replace the contents of the collection by clearing the collection first and then adding the new contents. If replacing the whole collection is a common scenario, consider providing the AddRange method on the collection.